Tools/TestWebKitAPI/Tests/WebKitCocoa/CookieStoreAPI.mm

```diff
@@ -61,4 +61,26 @@ HTTPServer server({
     EXPECT_TRUE([result isEqualToString:@"cookieValue"]);
 }
 
+TEST(WebKit, CookieStoreSetCookieForPublicSuffixDomain)
```

Comment:
Can't this be a layout test?

Reply:
The host is a layout test is "localhost". So if I try to set a cookie with a domain "com", the test will fail the domain-matching check before it gets to the public-suffix check.